test/backend/routes.tags.spec.js

```diff
@@ -0,0 +1,169 @@
+            .then(assertResponseStatus(200))
+            .then(getResponseJson)
+            .then(json => {
+                assert.deepEqual(json, {});
+            });
+    });
+
+    afterEach(() => {
+        return closeTestServers(servers);
+    });
+
+    describe('tags:', function() {
+        const TAG = {name: 'name', color: '#ffffff'};
+        const TAG_2 = {name: 'name2', color: '#000000'};
+
+        it('creates a tag', function() {
+            return POST('tags', TAG)
+                .then(assertResponseStatus(201))
+                .then(getResponseJson)
+                .then(json => {
+                    const {id} = json;
+                    assert.deepEqual(json, {...TAG, id});
+                });
+        });
+
+        it('fails to create a tag with duplicate name', () => {
+            return POST('tags', TAG)
+                .then(assertResponseStatus(201))
+                .then(() => POST('tags', TAG))
+                .then(assertResponseStatus(400));
+        });
+
+        it('updates a tag', () => {
+            let createdTag;
+            const UPDATED_COLOR = '#cccccc';
+
+            return POST('tags', TAG)
+                .then(assertResponseStatus(201))
+                .then(getResponseJson)
+                .then(json => {
+                    createdTag = json;
+                })
+                .then(() => PUT(`tags/${createdTag.id}`, {color: UPDATED_COLOR}))
+                .then(assertResponseStatus(200))
+                .then(getResponseJson)
+                .then(json => {
+                    assert.deepEqual(json, {...createdTag, color: UPDATED_COLOR});
+                });
+        });
+
+        it('fails to create a tag with invalid name or color parameters', () => {
+            const EMPTY_TAG = {};
+            const TAG_WITH_INVALID_NAME = {name: '_'.repeat(31), color: '#ffffff'};
+            const TAG_WITH_INVALID_COLOR = {name: 'name', color: '_'};
+
+            return POST('tags', EMPTY_TAG)
+                .then(assertResponseStatus(400))
+                .then(() => POST('tags', TAG_WITH_INVALID_NAME))
+                .then(assertResponseStatus(400))
+                .then(() => POST('tags', TAG_WITH_INVALID_COLOR))
+                .then(assertResponseStatus(400));
+        });
+
+        it('gets all tags', function() {
+            return POST('tags', TAG)
+                .then(assertResponseStatus(201))
+                .then(() => POST('tags', TAG_2))
+                .then(assertResponseStatus(201))
+                .then(() => GET('/tags'))
+                .then(assertResponseStatus(200))
+                .then(getResponseJson)
+                .then(json => {
+                    assert.equal(json.length, 2);
+                    assert.equal(json[0].name, TAG.name);
+                    assert.equal(json[1].name, TAG_2.name);
+                });
+        });
+
+        // TODO: run this test once tags are persisted on queries
```

Comment:
Could you please do a quick pass and make sure all TODOs are either deleted or moved to a separate issue (one issue with a checklist would be fine) ?

Reply:
Are you referring to TODOs from our work? I believe that was the only one and I removed it as it's outdated.

Reply:
cool, thanks! it's the only one I saw but I just wanted a check :)